target_chains/aptos/contracts/sources/pyth.move

```diff
@@ -163,20 +171,113 @@ module pyth::pyth {
         };
     }
 
-    fun update_price_feed_from_single_vaa(vaa: vector<u8>) {
-        // Deserialize the VAA
-        let vaa = vaa::parse_and_verify(vaa);
-
-        // Check that the VAA is from a valid data source (emitter)
+    fun verify_data_source(vaa: &vaa::VAA) {
         assert!(
             state::is_valid_data_source(
                 data_source::new(
-                    u16::to_u64(vaa::get_emitter_chain(&vaa)),
-                    vaa::get_emitter_address(&vaa))),
+                    u16::to_u64(vaa::get_emitter_chain(vaa)),
+                    vaa::get_emitter_address(vaa))),
             error::invalid_data_source());
+    }
+
+    fun parse_accumulator_merkle_root(message: vector<u8>): vector<u8> {
```

Comment:
please document these methods. it's not very clear what they do (especially since the signatures are all vec<u8>)